.circleci/config.yml

```diff
@@ -1,51 +1,130 @@
+# persist it to a workspace to be made available when the Jekyll site builds.
 references:
-  js_deps_cache_key: &js_deps_cache_key
-    v8-dependency-js-deps-{{ checksum "package-lock.json" }}
-  js_deps_backup_cache_key: &js_deps_backup_cache_key
-    v8-dependency-js-deps
   workspace_root: &workspace_root
     /tmp/workspace
   attach_workspace: &attach_workspace
     attach_workspace:
       at: *workspace_root
 
-jobs:
-  js_build:
-    docker:
-      - image: node:8.11.1
+
+# Several steps in this config use the same, specialized ruby-caching steps.
+# Commands can be used to extract a common set of steps into a reusable-block.
+# In this case, whenever you see `- ruby-deps` as a "step" in the config below,
+# It is executing a command to restore_cache, install deps if needed, and save_cache.
+commands:
+  ruby-deps:
+    description: "Runs specialized Ruby cache steps."
     steps:
-      - *attach_workspace
-      - checkout
       - restore_cache:
-          keys:
-            - *js_deps_cache_key
-            - *js_deps_backup_cache_key
-      - run:
-          name: npm dependencies
-          command: npm install
+          key: circleci-docs-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
       - run:
-          name: Prepare JS assets
+          name: Install Ruby dependencies
+          command: bundle check --path=vendor/bundle || bundle install --path=vendor/bundle --jobs=4 --retry=3
+      - save_cache:
+          key: circleci-docs-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
+          paths:
+            - "vendor/bundle"
+
+
+jobs:
+  js_build: # this job is responsible for building Javascript assets and making them available for the "build" job
+    executor: # we specify our executor to use the node orb.
+      name: node/default
+      tag: '8.11.1'
+    steps:
+      - *attach_workspace
+      - checkout # get the code from GitHub
+      - node/with-cache: # An orb command for steps to run in-between restoring and saving a cache.
+          cache-key: "package-lock.json"
+          cache-version: v9
+          steps:
+            - run: npm install
+      - run:
+          name: "Prepare JS assets" # Compile our final, production-ready JavaScript.
           command: npm run webpack-prod
       - run:
-          name: Persist JS assets
+          name: "Persist JS assets" # Move our JS into our workspace.
           command: |
             set -exu
             mkdir -p /tmp/workspace/js
             mv jekyll/assets/js/*.bundle.js /tmp/workspace/js/
-      - persist_to_workspace:
+      - persist_to_workspace: # store the built files into the workspace for other jobs.
           root: *workspace_root
           paths:
             - js
+
+  build_server_pdfs: # this job builds pdf versions of the Server Administration documentation and persists them to the workspace.
+    executor:
+      name: ruby/default # we use the same ruby orb + tag across multiple jobs
```

Comment:
Does it make sense to change to use this docker image for this job? https://hub.docker.com/r/asciidoctor/docker-asciidoctor